app/src/main/java/de/danoeh/antennapod/adapter/SubscriptionsRecyclerAdapter.java

```diff
@@ -0,0 +1,237 @@
+        private TriangleLabelView count;
+        private CheckBox selectCheckbox;
+        public SubscriptionViewHolder(@NonNull View itemView) {
+            super(itemView);
+            feedTitle = itemView.findViewById(R.id.txtvTitle);
+            imageView = itemView.findViewById(R.id.imgvCover);
+            count = itemView.findViewById(R.id.triangleCountView);
+            selectCheckbox = itemView.findViewById(R.id.selectCheckBox);
+        }
+
+        public void onBind(NavDrawerData.DrawerItem drawerItem, int position) {
+            feedTitle.setText(drawerItem.getTitle());
+            imageView.setContentDescription(drawerItem.getTitle());
+            feedTitle.setVisibility(View.VISIBLE);
+
+            if (TextUtilsCompat.getLayoutDirectionFromLocale(Locale.getDefault())
+                    == ViewCompat.LAYOUT_DIRECTION_RTL) {
+                count.setCorner(TriangleLabelView.Corner.TOP_LEFT);
+            }
+
+            if(!inActionMode())
+                if (drawerItem.getCounter() > 0) {
+                    count.setPrimaryText(NumberFormat.getInstance().format(drawerItem.getCounter()));
+                    count.setVisibility(View.VISIBLE);
+//                    count.
+                } else {
+                    count.setVisibility(View.GONE);
+                }
+            else {
+                if (isSelected(position)) {
+                    count.setPrimaryText("\u2713");
+                    count.setVisibility(View.VISIBLE);
+                } else {
+                    count.setVisibility(View.GONE);
+                }
+            }
+
+        if (drawerItem.type == NavDrawerData.DrawerItem.Type.FEED) {
+            Feed feed = ((NavDrawerData.FeedDrawerItem) drawerItem).feed;
+            boolean textAndImageCombind = feed.isLocalFeed()
+                    && LocalFeedUpdater.getDefaultIconUrl(itemView.getContext()).equals(feed.getImageUrl());
+            new CoverLoader(mainActivityRef.get())
+                    .withUri(feed.getImageUrl())
+                    .withPlaceholderView(feedTitle, textAndImageCombind)
+                    .withCoverView(imageView)
+                    .load();
+        } else {
+            new CoverLoader(mainActivityRef.get())
+                    .withResource(R.drawable.ic_folder)
+                    .withPlaceholderView(feedTitle, true)
+                    .withCoverView(imageView)
+                    .load();
+        }
+
+        if (inActionMode()) {
+            selectCheckbox.setVisibility(View.VISIBLE);
+            if (isSelected(position)) {
+                selectCheckbox.setChecked(true);
+            } else {
+                selectCheckbox.setChecked(false);
+            }
+            selectCheckbox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+                @Override
+                public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+                    setSelected(position, isChecked);
```

Comment:
What happens if you select a folder? I would argue that folders should either not be selectable at all or should add their whole content to the selection.

Reply:
I think multi-select should not be possible on the side menu – only in the Podcasts screen. And so, for now, selecting groups/tags wouldn't be possible either.

Reason is that there's no way to offer the multi-select functionality in a nice way on the side menu, and it's not the right place to manage individual podcasts (which is effectively what you're doing, when using multi-select).

Reply:
The subscriptions screen also shows folders currently

Reply:
> What happens if you select a folder? I would argue that folders should either not be selectable at all or should add their whole content to the selection.

The current implementation simply prevents a folder from being selected.